pkg/protocol/engine/syncmanager/trivialsyncmanager/syncmanager.go

```diff
@@ -126,26 +134,18 @@ func New(subModule module.Module, e *engine.Engine, latestCommitment *model.Comm
 }
 
 func (s *SyncManager) SyncStatus() *syncmanager.SyncStatus {
-	s.lastAcceptedBlockSlotLock.RLock()
-	s.lastConfirmedBlockSlotLock.RLock()
-	s.latestCommitmentLock.RLock()
-	s.latestFinalizedSlotLock.RLock()
-	s.lastPrunedEpochLock.RLock()
-	defer s.lastAcceptedBlockSlotLock.RUnlock()
-	defer s.lastConfirmedBlockSlotLock.RUnlock()
-	defer s.latestCommitmentLock.RUnlock()
-	defer s.latestFinalizedSlotLock.RUnlock()
-	defer s.lastPrunedEpochLock.RUnlock()
+	lastPrunedEpoch, hasPruned := s.LastPrunedEpoch()
 
 	return &syncmanager.SyncStatus{
-		NodeSynced:             s.IsNodeSynced(),
 		NodeBootstrapped:       s.IsBootstrapped(),
```

Comment:
This changes the locking behaviour actually. Because you don't read all the values once you got all locks, you instead read (and potentially wait) for each single value.